asv/environment.py

```diff
@@ -862,11 +863,39 @@ def _build_project(self, repo, commit_hash, build_dir):
         """
         Run build commands
         """
+        build_dir_path = Path(build_dir)
+
+        def has_file(file_name):
+            return (build_dir_path / file_name).exists()
+
         cmd = self._build_command
+
         if cmd is None:
-            cmd = ["python setup.py build",
-                   ("PIP_NO_BUILD_ISOLATION=false "
-                    "python -mpip wheel --no-deps --no-index -w {build_cache_dir} {build_dir}")]
+            if has_file('pyproject.toml'):
+                if has_file('pdm.lock'):
+                    cmd = [
+                        "python -mpip install pdm",
+                        "pdm install --prod",
+                        "pdm build --wheel --no-isolation --output {build_cache_dir}"
+                    ]
+                elif has_file('poetry.lock'):
+                    cmd = [
+                        "python -mpip install poetry",
+                        "poetry install",
+                        "poetry build --format wheel --output {build_cache_dir}"
```

Comment:
I am getting an error when running with poetry and python 3.10, `The option "--output" does not exist`, when either setting the build command in asv.conf.json or running this branch directly without a build command specified in the config. I don't see the `--output` option specified anywhere in poetry's (or pdm's) docs for the build command. Where does that option come from?

Reply:
It used to be passed `build` transparently at some point, however currently pending https://github.com/python-poetry/poetry/pull/7208 this needs to be reworked. Great catch!